index.html

```diff
@@ -134,247 +134,142 @@
-            to the current time.</li>
-        <li>If the <em>startMark</em> argument is specified, but the <em>endMark </em>argument is not specified,
-            <code>measure()</code> will store the duration as a <a href="http://www.w3.org/TR/hr-time-2/#domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-             from the most recent occurrence of the start mark to the current time.</li>
-        <li>If both the <em>startMark</em> and <em>endMark </em>arguments are specified, <code>measure()</code> will store the duration as a <a href="http://www.w3.org/TR/hr-time-2/#domhighrestimestamp"><code>DOMHighResTimeStamp</code></a> from the most recent occurrence of the start mark to the most recent occurrence of the end mark. The resulting duration value MAY be negative.</li>
-    </ul>
-    <p>The <em>startMark</em> and <em>endMark</em> arguments may be the name of one of the attributes in the
-       <code><a href="http://www.w3.org/TR/navigation-timing-2/#performancetiming">PerformanceTiming</a></code> interface [[!navigation-timing-2]].
+partial interface Performance {
+    void mark(DOMString markName);
+    void clearMarks(optional DOMString markName);
+
+    void measure(DOMString measureName, optional DOMString startMark, optional DOMString endMark);
+    void clearMeasures(optional DOMString measureName);
+};
+</pre>
+  <p>
+    All <a href='http://www.w3.org/TR/html51/webappapis.html#global-object'>global object</a> [[!HTML51]] have a <dfn>mark list</dfn>. Unless the <a for='Performance'>mark</a> is invoked, its <a>mark list</a> is empty. Each member of the list is a <a>PerformanceMark</a> object.
+  </p>
+
+  <p>
+    All <a href='http://www.w3.org/TR/html51/webappapis.html#global-object'>global object</a> have a <dfn>measure list</dfn>. Unless the <a for='Performance'>measure</a> is invoked, its <a>measure list</a> is empty. Each member of the list is a <a>PerformanceMeasure</a> object.
+  </p>
+
+  <p>The <dfn for="Performance" data-lt='mark'>mark(markName)</dfn> method stores a timestamp with the associated name (a "mark"). It MUST run these steps:</p>
+  <ol>
+    <li>If <var>markName</var> uses the same name as an attribute in the
+        <code><a href="http://www.w3.org/TR/navigation-timing-2/#idl-def-PerformanceTiming">PerformanceTiming</a></code> interface [[!NAVIGATION-TIMING-2]], <a href='https://heycam.github.io/webidl/#dfn-throw'>throw</a> a <a href='https://heycam.github.io/webidl/#syntaxerror'><code>SyntaxError</code></a>.</li>
+    <li>Create a new <a>PerformanceMark</a> object.</li>
+    <li>Set <code>name</code> to <var>markName</var>.</var>
+    <li>Set <code>entryType</code> to <code>DOMString "mark"</code>.
+    <li>Set <code>startTime</code> to the time in milliseconds from the <a href='http://w3c.github.io/hr-time/#time-origin'>time origin</a> to the occurrence of the call of the <a for='Performance'>mark</a> method.</li>

```

Comment:
Set <var>startTime</var> to the value that returned by the `Performance` object's `now()` method... ?